File: editor/en/[1614]Maximum Nesting Depth of the Parentheses.java
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int maxDepth(String s) {
        int current = 0;
        int max = 0;

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);

            if (c == '(') {
                current++;
                max = Math.max(max, current);
            } else if (c == ')') {
                current--;
            }
        }

        return max;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[1614]Maximum Nesting Depth of the Parentheses.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def maxDepth(self, s: str) -> int:
        current = 0
        max_length = 0

        for c in s:
            if c == '(':
                current += 1
                max_length = max(max_length, current)
            elif c == ')':
                current -= 1

        return max_length
# leetcode submit region end(Prohibit modification and deletion)
